test/corecel/cont/InitializedValue.test.cc

```diff
@@ -0,0 +1,129 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/cont/InitializedValue.test.cc
+//---------------------------------------------------------------------------//
+#include "corecel/cont/InitializedValue.hh"
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+
+TEST(InitializedValue, no_finalizer)
+{
+    using InitValueInt = InitializedValue<int>;
+    // TODO: in C++20 use [[no_unique_address]] and restore this assertion
+    // static_assert(sizeof(InitValueInt) == sizeof(int), "Bad size");
+
+    // Use operator new to test that the int is being initialized properly by
+    // constructing into data space that's been set to a different value
+    alignas(int) Byte buf[sizeof(int)];
+    std::fill(std::begin(buf), std::end(buf), Byte(-1));
+    InitValueInt* ival = new (buf) InitValueInt{};
```

Comment:
error: placement new constructing an object of type ‘InitValueInt’ {aka ‘celeritas::InitializedValue<int>’} and size ‘8’ in a region of type ‘celeritas::Byte [4]’ and size ‘4’ [-Werror=placement-new=]
   28 |     InitValueInt* ival = new (buf) InitValueInt{};